2week/src/question/components/one-todo.tsx

```diff
@@ -0,0 +1,30 @@
+import type { TodoDataBase } from '@/types/todo'
+import { TodoEnum } from '../../types/todo'
+interface Props {
+  todo: TodoDataBase
+}
+
+const OneTodo: React.FC<Props> = ({ todo }) => {
+  if (todo.type === TodoEnum.DAILY) {
+    return (
+      <div>
+        <h3>{todo.title}</h3>
+        <div>{todo.content}</div>
+        <hr />
+      </div>
+    )
+  }
+
+  if (todo.type === TodoEnum.WEEKLY) {
+    return (
+      <div>
+        <h3>{JSON.stringify(todo.total)}</h3>
+      </div>
+    )
+  }
+
```

Comment:
if 문에 따라 다른 html을 return 하는 것은 유지 보수에 좋은 패턴이 아닙니다.

타입가드 함수와 함께 useMemo나 즉시실행 함수를 사용하는 것이 좋습니다.
```suggestion

const todotypeisDatily = (todo:TodoDatBase): todo is TodoType<TodoEnum.DAILY> => todo.type === TodoEnum.DAILY;

const TODO = (()=>{
 if(todotypeisDatily(todo)){
   return {
      title,
      content,
      goal
   }
 }
})()

<div>{Todo.title}</div>
```

Reply:
[refactor: 즉시실행함수 + 타입가드 활용](https://github.com/mobi-community/mobi-path-typescript/pull/3/commits/acb210761bec7c8a0faea727ed05d116713a2da0)

리뷰해주신 내용 참고하여 수정했습니다 👍 즉시실행함수 활용 흥미로웠습니다 🫢